File: java/src/DirectionNorth.java

public class DirectionNorth implements IDirection
{
	public IDirection left()
	{
		return new DirectionWest();
	}
	
	public IDirection right()
	{
		return new DirectionEast();
	}
	
	public Position move(Position position)
	{
		position.setYPosition(position.getYPosition() + 1 );
		return position;
	}

	public String name()
	{
		return "NORTH";
	}
}

File: ruby/src/DirectionNorth.rb
require_relative "IDirection.rb"
require_relative "DirectionWest.rb"
require_relative "DirectionEast.rb"

class DirectionNorth < IDirection
  
  def name()
    return "NORTH"
  end
  
  def left()
    return DirectionWest.new()
  end
  
  def right()
    return DirectionEast.new()
  end
  
  def move(position)
    position.y_position = position.y_position + 1
    return position
  end
end
